cdap-common/bin/functions.sh

```diff
@@ -758,7 +761,11 @@ cdap_start_java() {
   eval split_jvm_opts ${!JAVA_OPTS_VAR} ${OPTS} ${JAVA_OPTS}
   local __defines="-Dcdap.service=${CDAP_SERVICE} ${JAVA_HEAPMAX} -Duser.dir=${LOCAL_DIR} -Djava.io.tmpdir=${TEMP_DIR}"
   # Enable GC logging
-  __defines+=" -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${__log_prefix}-heap.hprof -verbose:gc -Xloggc:${__log_prefix}-gc.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=1M"
+  cdap_create_dir ${__gc_log_and_heapdump_dir}
+  if [ "$HEAPDUMP_ON_OOM" = true ] ; then
```

Comment:
[[ ]] preferred per style guide: https://google.github.io/styleguide/shell.xml, but you also need ``==``, and variable brace style